ch19-06 typo “this writing” -> “writing this” (#3481)
rust-lang/book: src/ch19-06-macros.md

on: Rust doesn’t have reflection capabilities, so it can’t look up the type’s
name at runtime. We need a macro to generate code at compile time.

> The next step is to define the procedural macro. At the time of writing this,

Reviewer: Perhaps you can consider eliminating "this" entirely, shortening it to:
> At the time of writing,

There's an [english stackexchange post](https://english.stackexchange.com/questions/199381/is-at-the-time-of-writing-correct) about this.

Author: I agree - it does sound better

> The next step is to define the procedural macro. At the time of writing,
> procedural macros need to be in their own crate. Eventually, this restriction
> might be lifted. The convention for structuring crates and macro crates is as

Reviewer 2: ```suggestion
The next step is to define the procedural macro. Due to technical limitations they must be in their own crate. The convention for structuring crates and macro crates is as
```

What about just getting rid of it? There's no concrete plan or rfc for doing this. Saying "might be lifted" means nothing.

Author: If there is no rfc for this then removing this is a good idea.

Reviewer 3: > If there is no rfc for this then removing this is a good idea.

Actually it does to me. It conveys the fact that it is this way now but may not always be this way. Saying due to technical limitations is similar but doesn't cover the fact that that limitation is not necessarily always going to exist. While one can assume that it may not exist one day the former way clearly indicates that.

Author: That is a valid point of view. However I believe that anyone that is reading this book is reading it to lean the current rust. Sure, this might change in the future but then we can just update this. It is obvious that rust is going to evolve and change.

Reviewer 2: Some people definitely read it as "work is being done on it". For example:

![image](https://user-images.githubusercontent.com/59372212/210177664-1858ef13-425f-4e39-b41d-148e3047b06d.png)
